src/__tests__/config-snapshot.test.ts

```diff
@@ -0,0 +1,54 @@
+import ts from 'typescript'
+
+function extractTypeInfo(filePath: string, typeName: string): string {
+    const program = ts.createProgram([filePath], {})
+    const checker = program.getTypeChecker()
+    const sourceFile = program.getSourceFile(filePath)
+
+    if (!sourceFile) {
+        throw new Error(`File not found: ${filePath}`)
+    }
+
+    function getTypeString(type: ts.Type): string {
+        return checker.typeToString(type)
+    }
+
+    function processType(type: ts.Type): any {
+        if (type.isUnion() || type.isIntersection()) {
+            return type.types.map(getTypeString)
```

Comment:
Rather than calling `getTypeString` we should `processType` recursively, this will allow us to capture nested unions like `{a: {b: number}} | {a: {b: string}}`

Reply:
yep, and that picked up things like AutocaptureConfig without them needing a separate test 👍